### PR TITLE
feat(helm): add priorityClass and extraLabels to kube-agent

### PR DESCRIPTION
**Pull request by daquinoaldo**
- Allows setting a `priorityClass` (as in `teleport-cluster`).
- Allow setting labels on Kubernetes resources via `extraLabels` (similarly to how we set `annotations`).

cc @webvictim

### CODE REVIEW

File: docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx [hunk 1]
```diff
@@ -768,6 +768,27 @@ Ensures that this number of replicas is available during voluntary disruptions,
...
+
+<Tabs>
+  <TabItem label="values.yaml">
+  ```yaml
+  priorityClassName: "teleport-kube-agent"
+  ```
+  </TabItem>
+  <TabItem label="--set">
+  ```code
+  $ --set serviceAccountName=teleport-kube-agent
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx:787]:
```suggestion
  $ --set priorityClassName=teleport-kube-agent
```

File: examples/chart/teleport-kube-agent/values.schema.json [hunk 2]
```diff
@@ -253,6 +264,63 @@
...
             "default": {}
         },
+        "extraLabels": {
+            "$id": "#/properties/extraLabels",
+            "type": "object",
+            "required": [
+                "config",
+                "deployment",
+                "pod",
+                "serviceAccount"
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.schema.json:274]: I don't think these are actually required? All `extraLabels` are inherently optional, right, so there shouldn't be any required parameters.

**Review by webvictim — COMMENTED** [review]
LGTM with a couple of changes required. I'm going to run the tests.

Also, it looks like you need to run `git submodule update webassets` in your Teleport git checkout to make sure it's in sync with the upstream version, as there's an unrelated submodule change proposed here.

Thanks for the PR!

**Comment by webvictim** [comment]
The work here does seem to cross over with what @Aharic is doing in https://github.com/gravitational/teleport/pull/12160 - tagging him so he can be aware. I actually prefer the implementation in this PR as it allows you to only set labels on specific elements of the chart.

**Comment by r0mant** [comment]
@daquinoaldo Thank you very much for the contribution! I've opened a buddy PR here: https://github.com/gravitational/teleport/pull/12559, with @webvictim's feedback applied. We'll get that merged and released.

**Pull request closed without merging**